### PR TITLE
Add specs for Enumerable#slice_when

### PR DESCRIPTION
**Pull request by jsyeo**
:sweat_smile:

### CODE REVIEW

File: spec/ruby/core/enumerable/slice_when_spec.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  context "when given a block" do
+    it "returns an enumerator" do
+      @result.should be_an_instance_of(enumerator_class)
+    end
+
+    it "splits chunks between adjacent elements i and j where the block returns true" do
+      @result.to_a.should == [[10, 9], [7, 6], [4, 3, 2, 1]]
+    end
+
+    it "calls the block the length of the receiver enumerable minus one times" do
```
Comment (YorickPeterse) [inline spec/ruby/core/enumerable/slice_when_spec.rb]: What is "calls the block the length" supposed to mean?

File: spec/ruby/core/enumerable/slice_when_spec.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      times_called = 0
+      @enum.slice_when do
+        times_called += 1
+        false
```
Comment (YorickPeterse) [inline spec/ruby/core/enumerable/slice_when_spec.rb]: Any particular reason why this block is returning `false`?

File: spec/ruby/core/enumerable/slice_when_spec.rb (hunk 1 above)
Comment (jsyeo) [inline spec/ruby/core/enumerable/slice_when_spec.rb]: Should be calls the block for ... times :confounded: . I wasn't particularly awake at the time I wrote it. I have fixed it in the latest commit.

File: spec/ruby/core/enumerable/slice_when_spec.rb (hunk 2 above)
Comment (jsyeo) [inline spec/ruby/core/enumerable/slice_when_spec.rb]: Alright, I think it's a bad idea to use `false`. I've decided to stick to my `i - 1 != j` example in my latest commit.

**Comment by YorickPeterse** [comment]
Github is now showing 2 commits with the same message, both (as it appears) modifying the same file. Could you rebase these commits into a single one?

**Comment by jsyeo** [comment]
:confused: not sure how that happened. I probably edited the commit message instead of squashing it. Anyway, I've fixed it.

**Comment by YorickPeterse** [comment]
:+1:

**Pull request merged**